src/TreeContainer.js

```diff
@@ -243,11 +168,78 @@ function mergeProps(stateProps, dispatchProps, ownProps) {
         _dashprivate_dependencies: stateProps.dependencies,
         _dashprivate_dispatch: dispatchProps.dispatch,
         _dashprivate_layout: ownProps._dashprivate_layout,
+        _dashprivate_loadingState: getLoadingState(ownProps._dashprivate_layout, stateProps.requestQueue),
```

Comment:
Calculating the loadingState upfront instead of in the render function -- this allows to block/allow re-renderer based on loading state in `shouldComponentUpdate`